### PR TITLE
Support for Building on Fedora 21

### PR DESCRIPTION
**Pull request by DavidGosselin**
Adds support for compiling on Fedora 21 by enabling the `make deps` target to build successfully.

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: tools/provision/fedora.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  package gcc-c++
+  package clang
+  package clang-devel
+
+  install_cmake
+
+  set_cc clang
+  set_cxx clang++
+
+  install_boost
```
Comment (theopolis) [inline tools/provision/fedora.sh]: The boost-devel package doesn't work? Perhaps it doesn't install static libs?

File: tools/provision/fedora.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  package byacc
+  package flex
+  package bison
+  package autoconf
+  package automake
+  package libtool
+
+  install_snappy
+  install_rocksdb
+  install_thrift
```
Comment (theopolis) [inline tools/provision/fedora.sh]: It looks like there's a thrift-devel package, but it doesn't provide the compiler, only headers/libs. :(

File: tools/provision/fedora.sh (hunk 1 above)
Comment (theopolis) [inline tools/provision/fedora.sh]: Right, only shared libs in that package.

File: tools/provision/fedora.sh (hunk 2 above)
Comment (DavidGosselin) [inline tools/provision/fedora.sh]: Is there a change that you'd like me to make?

**Comment by theopolis** [comment]
This looks great! You'll need to use our standard copyright header and accept the Facebook CLA as per the friendly bot's message above. As soon as the bot receives your CLA confirmation we can merge this!

**Comment by DavidGosselin** [comment]
Awesome, I've signed the corporate CLA, with authorization from my employer, for this contribution, adding myself as an authorized contributor.  The copyright header follows the standard format with our company name as the exception. As this contribution is BSD licensed, along with the CLA in place, I don't understand why I need to change the company name in the fedora.sh file.  Can you please clarify?  Thanks.

**Comment by theopolis** [comment]
I can try my best; the CLA provides that contributions to Facebook-managed projects/code are licensed by Facebook, in this case under a BSD with patent caveats. However, it also provides that copyright of those contributions is given to Facebook. But submitting a PR, signing the CLA, and a maintainer merging the code, the copyright owner and licensee becomes Facebook. 

That said, we're looking for ways to track our contributiors and community maintainers. No one has made a PR for a COMTRIBUTORS.md yet, until then (or an alternative) I'm a huge fan of the Github provided contributors/modifications history.

**Comment by DavidGosselin** [comment]
Should be all set, thank you.

**Comment by theopolis** [comment]
ok to test

**Comment by theopolis** [comment]
Hey @DavidGosselin, I don't see a signed CLA from your Github account. Did you complete the form: https://code.facebook.com/cla using a different Github username?

**Comment by DavidGosselin** [comment]
Yes, I signed the CLA and have a printed copy here that indicates that I have successfully signed the CLA.  I signed a corporate one for my company as indicated before and have linked my GitHub account to it as one of the designated employees.

**Comment by theopolis** [comment]
The email linking might be the issue. I tried to search by company and cannot find 128 Technologies either. Would you mind sending me an email: `reed@fb.com` from the email account you signed with and I'll look for a signed copy on my end.

**Comment by jamesgpearce** [comment]
CLA good to go - thanks!

**Comment by theopolis** [comment]
Ok perfect! Thanks @jamesgpearce, and thanks @DavidGosselin!

**Comment by theopolis** [comment]
@DavidGosselin, do you mind rebasing to master? There was an update to `lib.sh` I believe. Also, do you mind squashing your commits into 1? `git rebase -i HEAD~3`, then change 'pick' to 's' for the second and third commits.

**Comment by DavidGosselin** [comment]
The rebase doesn't appear to have worked properly, I need to update the commit before you can pull it in.

**Comment by theopolis** [comment]
Looks great!

**Comment by DavidGosselin** [comment]
Thanks!  I think it's all set for merging upstream, thanks again for your help.

**Pull request merged**